### ISSUE TITLE
`define-macros-order`: should ignore `declare` TypeScript declarations along with type definitions

### ISSUE THREAD

**Issue by andreww2012** [vuejs/eslint-plugin-vue#2590]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 9.13.0
- **eslint-plugin-vue version:** 9.30.3
- **Vue version:** 3.5.12
- **Node version:** 
- **Operating System:** 

**What did you do?**	

Since `declare` TypeScript declarations only exist at compile time, I think they should be allowed to be placed before macros along with type definitions, meaning that the following code should not be reported:

```vue
<script setup lang="ts">
declare global {};

defineProps(/* ... */)
defineEmits(/* ... */)
</script>
```

[Test on the rule page](https://eslint.vuejs.org/rules/define-macros-order.html)

**What did you expect to happen?**

**What actually happened?**

**Repository to reproduce this issue**
